### PR TITLE
added sem ver clarification - #850

### PR DESCRIPTION
**Pull request by cholmes**
**Related Issue(s):** #850 

**Proposed Changes:**

1. clarify we use sem ver

**PR Checklist:**

- [x] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [x] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.

### CODE REVIEW

**Comment by cholmes** [comment]
@m-mohr - let me know if this works, happy to tweak more.

File: README.md [hunk 1]
```diff
@@ -23,7 +23,8 @@ in a 'beta' state, with no major changes anticipated, so implementors can expect
 at this point we don't anticipate any major changes, but reserve the right to make them if we get feedback that something just doesn't 
 work. Which is to say the next couple months are a great time to implement STAC, as your changes will be head. After 1.0 our goal
 is to not change the core in any backwards incompatible way for a very long time, if ever, so that people can build on this until 
-JSON is no longer relevant. 
+JSON is no longer relevant. STAC spec aims to follow [Semantic Versioning](https://semver.org/), so once 1.0.0 is reached any breaking
```
Comment (m-mohr) [inline README.md:26]:
```suggestion
JSON is no longer relevant. The STAC specification follows [Semantic Versioning](https://semver.org/), so once 1.0.0 is reached any breaking
```

Comment (cholmes) [reply README.md:26]: The reason I said 'aims' is that I feel like pre-1.0 we haven't perfectly followed it. Though I guess this commit is just 'going forward', and our 1.0.0-beta.1 does follow it.

Comment (m-mohr) [reply README.md:26]: I think SemVer doesn't make any assumptions on breaking changes for 0.x versions (you can do whetever you want), so we are totally fine with the proposal.

Comment (m-mohr) [reply README.md:26]: Yep, proof is here: https://semver.org/#spec-item-4

Comment (cholmes) [reply README.md:26]: Cool.

**Pull request merged**